docs/platform-specification.md

```diff
@@ -681,9 +761,112 @@ Other upload parameters can also be defined for the board. For example, in the A
+    {upload.port.properties.name} = ttyACM0
+    {serial.port} = /dev/ttyACM0                # for backward compatibility
+    {serial.port.file} = ttyACM0                # only because protocol=serial
+
+Here another example:
+
+    {
+    "eventType": "add",
+    "port": {
+        "address": "192.168.1.232",
+        "label": "SSH on my-board (192.168.1.232)",
+        "protocol": "ssh",
+        "protocolLabel": "SSH Network port",
+        "properties": {
+        "macprefix": "AA:BB:CC",
+        "macaddress": "AA:BB:CC:DD:EE:FF"
+        }
+    }
+    }
+
+that is translated to:
+
+    {upload.port.address} = 192.168.1.232
+    {upload.port.label} = SSH on my-board (192.168.1.232)
+    {upload.port.protocol} = ssh
+    {upload.port.protocolLabel} = SSH Network port
+    {upload.port.properties.macprefix} = AA:BB:CC
+    {upload.port.properties.macaddress} = AA:BB:CC:DD:EE:FF
+    {serial.port} = 192.168.1.232                  # for backward compatibility
+
+This configuration, together with protocol selection, allows to remove the hardcoded `network_pattern`, now we can
+replace the legacy recipe (split into multiple lines for clarity):
+
+    tools.bossac.upload.network_pattern="{runtime.tools.arduinoOTA.path}/bin/arduinoOTA"
+                                        -address {serial.port} -port 65280
+                                        -sketch "{build.path}/{build.project_name}.bin"
+
+with:
+
+    tools.arduino_ota.upload.pattern="{runtime.tools.arduinoOTA.path}/bin/arduinoOTA"
+                                    -address {upload.port.address} -port 65280
+                                    -sketch "{build.path}/{build.project_name}.bin"
+
+#### User provided fields
+
+Some upload recipes might require custom fields that must be provided by the user, like username and password to upload
+over the network. In this case the recipe must use the special placeholder {upload.field.FIELD_NAME} where FIELD_NAME
+must be declared separately in the recipe using the following format:
+
+    tools.UPLOAD_RECIPE_ID.upload.field.FIELD_NAME=FIELD_LABEL
+    tools.UPLOAD_RECIPE_ID.upload.field.FIELD_NAME.secret=true
+
+FIELD_LABEL is the label shown in the graphical prompt to ask the user to enter the value of the field. The secret
+property is optional and it should be set to true if the field is a secret (like passwords or token).
+
+Let’s see how a complete example will look like:
+
+    tools.arduino_ota.upload.field.username=Username
+    tools.arduino_ota.upload.field.password=Password
+    tools.arduino_ota.upload.field.password.secret=true
+    tools.arduino_ota.upload.pattern="{runtime.tools.arduinoOTA.path}/bin/arduinoOTA"
+                                    -address {upload.port.address} -port 65280
+                                    -username "{upload.field.username}
+                                    -password "{upload.field.password}"
+                                    -sketch "{build.path}/{build.project_name}.bin"
```

Comment:
```suggestion
    tools.arduino_ota.upload.pattern="{runtime.tools.arduinoOTA.path}/bin/arduinoOTA" -address {upload.port.address} -port 65280 -username "{upload.field.username} -password "{upload.field.password}" -sketch "{build.path}/{build.project_name}.bin"
```
It was reasonable in the special case where you wanted to make it easy to see the before and after and made an explicit note about it being broken for clarity, but I think it will cause a lot of confusion to platform authors to do that in other parts of the specification, since the properties data format doesn't provide any way to break things into multiple lines AFAIK.